docs-src/cloud/limits.md

```diff
@@ -169,7 +169,42 @@ These operations create Events that are then logged in the Event History.
 The transaction size limit restricts the total size of Events that can be accommodated within a single transaction.
 
 The size limit for any given [Event History](/concepts/what-is-an-event-history) transaction is 4 MB.
-This isn't configurable.
+This limit is non-configurable for Temporal Cloud.
+
+**What is the blob size limit for a transaction?**
+
+- **Blob size limit** for Payloads, including Workflow context and each Workflow and Activity argument and return value:
+  - Temporal warns at 256 KB: `Blob size exceeds limit`.
+  - Temporal errors at 2 MB: `ErrBlobSizeExceedsLimit: Blob data size exceeds limit.`
+
+This limit is non-configurable for Temporal Cloud.
+
+The following provides solutions for handling large payloads.
+
+**Solutions for handling large payloads**
```

Comment:
The solutions proposed here are not only applicable to cases where one would face blob-size errors, but also may be pertinent to cases of exceeding per-workflow limits.

For example:
- payload compression (and payload offloading, as suggested later) might help in cases of workflow size being too large; 
- smaller batches may help in cases of exceeding the number of concurrent ScheduleActivityTask/SignalExternalWorkflowExecution/StartChildWorkflowExecution/RequestCancelExternalWorkflowExecution.

These solutions are also pertinent in self-hosted scenarios.

Reply:
For those reasons, you may want to consider wither these propositions would better fit as a distinct page, linked from here.